### PR TITLE
Remove deprecated S3 settings

### PR DESCRIPTION
**Pull request by rjernst**
These settings are deprecated in 5.5. This change removes them for 6.0.

### CODE REVIEW

File: plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3ClientSettings.java [hunk 1]
```diff
@@ -81,11 +81,11 @@
 
     /** The number of retries to use when an s3 request fails. */
     static final Setting.AffixSetting<Integer> MAX_RETRIES_SETTING = Setting.affixKeySetting(PREFIX, "max_retries",
-        key -> Setting.intSetting(key, S3Repository.Repositories.MAX_RETRIES_SETTING, 0, Property.NodeScope));
+        key -> Setting.intSetting(key, 3, 0, Property.NodeScope));
```
Comment (dadoonet) [inline plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3ClientSettings.java]: I wonder if we should default to `ClientConfiguration.DEFAULT_RETRY_POLICY.getMaxErrorRetry()` instead?
Or to `PredefinedRetryPolicies.DEFAULT_MAX_ERROR_RETRY`?

File: plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3Repository.java [hunk 2]
```diff
@@ -58,149 +58,67 @@
      * NOTE: These are legacy settings. Use the named client config settings above.
      */
     public interface Repositories {
```
Comment (dadoonet) [inline plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3Repository.java]: Not part of this change but you can remove `public` here. Also for `public static final String TYPE = "s3";`

File: plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3Repository.java [hunk 3]
```diff
@@ -244,13 +162,8 @@ The default behaviour is to detect which access style to use based on the config
...
+        String basePath = BASE_PATH_SETTING.get(metadata.settings());
         if (Strings.hasLength(basePath)) {
-            if (basePath.startsWith("/")) {
-                basePath = basePath.substring(1);
-                deprecationLogger.deprecated("S3 repository base_path trimming the leading `/`, and " +
-                                                 "leading `/` will not be supported for the S3 repository in future releases");
-            }
             this.basePath = new BlobPath().add(basePath);
         } else {
             this.basePath = BlobPath.cleanPath();
```
Comment (dadoonet) [inline plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3Repository.java]: Not related to this change but method `public static <T> T getValue(...)` at the end of this class can become package private.

**Review by dadoonet — APPROVED** [review]
I left some minor comments. LGTM and nice to see all that code lighter than it was!

If you plan to add something in the breaking changes, you can may be take some of the stuff I initially wrote at https://github.com/elastic/elasticsearch/pull/23276/files#diff-c710699b84701013d83a5fe7ddc9a6d3.

**Comment by rjernst** [comment]
Thanks for the review @dadoonet. I pushed new commits with cleanups as you suggested, and also migration docs. Let me know if you are happy with the docs.

**Comment by dadoonet** [comment]
Great! Thanks!

**Comment by rjernst** [comment]
@elasticmachine retest this please

**Comment by rjernst** [comment]
> If you plan to add something in the breaking changes

The removal of the old settings are already listed in the breaking changes, when I made the PRs to remove them from master.
